keps/sig-node/4639-oci-volume-source/README.md

```diff
@@ -887,6 +916,16 @@ enhancement:
   CRI or CNI may require updating that component before the kubelet.
 -->
 
+Same as above: all the components must have support for it to work.
+Specifically, kube-apiserver will filter this field if it doesn't recognize/support it.
+If the kubelet doesn't support the field, it will not pull the image, and the path will not be present for the volume manager
+to mount it in, and the pod will fail.
+if the CRI implementation doesn't support it, then the PullImage request will fail and the pod creation will fail.
+
+If the kubelet does not support the feature because it's too old, then the
+container creation will fail because the volume manager is unable to attach or
+mount the volume because no volume plugin is available for the provided source.
```

Comment:
volume plugin?

Reply:
The image volume plugin is a no-op plugin. Before introducing the image volume type, the volume plugin did all operations on a volume. To avoid breaking the existing logic, the image volume plugin was introduced to handle the image volume type. However, most of the operations do nothing except for `GetAttributes` and `CanSupport` because the real operations are done by the ContainerRuntime.

https://github.com/kubernetes/kubernetes/blob/master/pkg/volume/image/image.go#L30

Reply:
@haircommander In most cases, attach/detach is handled by kcm. Image volume plugin does not support attach/detach.

Reply:
cc @saschagrunert

Reply:
@saschagrunert Can we address this if not resolved?

Reply:
Updated the sentence to not mention "attach". It was mostly a copy/paste from the error message users see when they try the skew scenario and the kubelet is too old.